addon/components/cfb-form-editor/question.js

```diff
@@ -292,26 +299,18 @@ export default Component.extend({
 
   submit: task(function* (changeset) {
     try {
-      const slug = ((!this.slug && this.prefix) || "") + changeset.get("slug");
-
       yield this.saveOptions.perform(changeset);
 
       const question = yield this.apollo.mutate(
         {
           mutation: TYPES[changeset.get("__typename")],
           variables: {
-            input: Object.assign(
-              {
-                label: changeset.get("label"),
-                slug,
-                isHidden: changeset.get("isHidden"),
-                infoText: changeset.get("infoText"),
-                meta: JSON.stringify(changeset.get("meta").unwrap()),
-                isArchived: changeset.get("isArchived"),
-                clientMutationId: v4(),
-              },
-              this[`_get${changeset.get("__typename")}Input`](changeset)
-            ),
+            input: {
+              ...this._getBaseInput(changeset),
+              ...(
+                this[`_get${changeset.get("__typename")}Input`] || (() => ({}))
```

Comment:
:laughing: this line was  already funny before, and this change makes it even better!

For the more casual reader it might be more friendly to use a variable assignment though.

Reply:
Fair

Reply:
Done